veracode/submit_artifacts.py

```diff
@@ -390,3 +360,67 @@ def submit_artifacts(  # pylint: disable=too-many-return-statements, too-many-br
         return False
 
     return True
+
+
+@validate
+def is_ready_for_new_build(*, upload_api: UploadAPI) -> bool:
+    """
+    Return whether a new build can be started
+
+    https://help.veracode.com/reader/orRWez4I0tnZNaA_i0zn9g/Yjclv0XIfU1v_yqmkt18zA
+    """
+    try:
+        endpoint = "getbuildinfo.do"
+        params = {"app_id": upload_api.app_id}
+
+        # If a sandbox_id is specified, add it to the params
+        if isinstance(upload_api.sandbox_id, str):
+            params["sandbox_id"] = upload_api.sandbox_id
+
+        response = upload_api.http_get(endpoint=endpoint, params=params)
+
+        if element_contains_error(parsed_xml=response):
```

Comment:
It looks like this is true when a build doesn't exist for the app/sandbox, but that is OK/expected especially for new sandboxes that we just created.  Example response:
```
b'<?xml version="1.0" encoding="UTF-8"?>\n\n<error>Could not find a build for application&#x3d;111111 and sandbox&#x3d;2111117</error>\n'
```
Is it possible we should be using `getbuildlist.do` instead of `getbuildinfo.do`?  I did a quick test, and it doesn't return an error if no builds exist.

Reply:
This approach is working well in my tests:
1) Run `getbuildlist.do` first to check if a sandbox with the same name exists
2) If true, proceed to `getbuildinfo.do` to check the status
3) If false, return early with `True` (build can be started)

I haven't seen any errors so far.

Reply:
Confirmed this works when the sandbox is first created in the Veracode UI.

Reply:
Working on handling a bad state that the above case doesn't handle. If a new sandbox ID that gets cancelled during the file upload stage, the sandbox is created, but does not show up in `getbuildlist.do`.

Reply:
Updated this to be much simpler by only using `getbuildlist.do` and only checking for an existing `build_id` to be associated with either the sandbox or policy scan. If one exists, we cancel it and redo the build. This completely ignores the complex parsing associated with `getbuildinfo.do`. So far no issues observed while testing.